packages/abstract/src/core/entities/entity/entity.ts

```diff
@@ -38,7 +38,7 @@ export class Entity<
     this.effects = () => [
       () => {
         // Re-run this effect whenever the `id` changes
-        const {id: _, manager} = this;
+        const {id, manager} = this;
```

Comment:
weren't we discarding the value of id from this here? (by assigning to _)

Reply:
The difference is when the effect runs again, it will use whatever id is set in `this`, instead of the `id` that was received by the class constructor. That way if the id changes you're using the new id, not the initially set one.